flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/test/util/MultipleProgramsTestBase.java

```diff
@@ -80,29 +80,36 @@
 	
 	protected final TestExecutionMode mode;
 
-	
+	private TestEnvironment testEnvironment;
+
+	private CollectionTestEnvironment collectionTestEnvironment;
+
 	public MultipleProgramsTestBase(TestExecutionMode mode) {
 		this.mode = mode;
-		
+
 		switch(mode){
 			case CLUSTER:
-				new TestEnvironment(cluster, 4).setAsContext();
+				testEnvironment = new TestEnvironment(cluster, 4);
```

Comment:
Thank you for responding. Last my comment is not correct. Actually what I want to say is that we can not move these code to \@BeforeClass and \@AfterClass. But we can move them to \@Before and \@After, however it will bring a little overhead since \@Before and \@After will be called for each test method. I pushed a new commit using \@Before and \@After.